Add attribute to able to configure custom theme

zeallat opened the pull request:
modify

Commit:
modify

modify

WindSekirun commented on demo/src/main/java/pyxis/uzuki/live/naraeimagepickerdemo/MainActivity.java:
```diff
@@ -59,6 +59,7 @@ private void clickSelect() {
         item.setViewMode(ViewMode.FolderView);
         item.setEnableUpInParentView(true);
         item.setDisableZoomMode(true);
+//        item.setThemeResId(R.style.CustomPickerTheme);
```
Please add sample theme style or remove comment.

WindSekirun commented on library/src/main/AndroidManifest.xml:
```diff
@@ -10,9 +10,9 @@
 
         <activity android:name=".activity.NaraePickerActivity"
-            android:theme="@style/Theme.AppCompat.Light.DarkActionBar"/>
+            android:theme="@style/NaraeImagePickerTheme"/>
```
When applying a style in Activity by programmatically, you don't need ```NaraeImagePickerTheme``` and extending that style. I can produce the same effort by this statement.

```
<style name="CustomPickerTheme" parent="Theme.AppCompat">
        <item name="colorPrimary">@color/colorPrimary</item>
        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
        <item name="colorAccent">@color/colorAccent</item>
    </style>
```

zeallat replied:
@WindSekirun 
I was add that theme for two reasons.
1. Refactor code
2. Guide user that what should be parent of their custom theme. (Strictly, not should be but for help users)

If you still feel this is not necessary, then I will remove it.

WindSekirun replied:
@zeallat 

Although AppCompat is used heavily, recent [MaterialComponent](https://material.io/develop/android/docs/getting-started/) should use ```Theme.MaterialComponent``` theme rather than ```Theme.AppComapt```. If so, at least this library should provide a base style for:

- Theme.AppCompat.Light.DarkActionBar
- Theme.AppCompat
- Theme.AppCompat.DayNight
- Theme.MaterialComponents
- Theme.MaterialComponents.Light.DarkActionBar

And I think many apps already have their own styles defined. So that's no problem to use self-defined style rather than base style.

zeallat replied:
@WindSekirun LGTM

WindSekirun commented on library/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="NaraeImagePickerTheme" parent="@style/Theme.AppCompat.Light.DarkActionBar"/>
```
Same as above.

WindSekirun commented on library/src/main/java/pyxis/uzuki/live/naraeimagepicker/utils/ActivityEx.kt:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Apply [PickerSettingItem] theme into Activity if configured.
+ */
+fun AppCompatActivity.applyCustomPickerTheme(settingItem: PickerSettingItem) {
```
should be 'internal' modifier cause child module can use this code.

zeallat replied:
@WindSekirun Ok. You mean that we have to restrict user can't access this function. Am I right?

WindSekirun replied:
:+1:

Reference: https://kotlinlang.org/docs/reference/visibility-modifiers.html#packages

WindSekirun commented on library/src/main/java/pyxis/uzuki/live/naraeimagepicker/activity/NaraePickerActivity.kt:
```diff
@@ -32,20 +33,21 @@ class NaraePickerActivity : AppCompatActivity() {
         EventBus.getDefault().register(this)
         RPermission.instance.checkPermission(this, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE), { integer, _ ->
+                Manifest.permission.WRITE_EXTERNAL_STORAGE)) { integer, _ ->
```
:+1:

WindSekirun reviewed (changes requested):
Looking good to me. Overall, you don't need to make 'NaraeImagePickerTheme' style for this feature. Please consider remove that style.

WindSekirun commented:
Please remove 'WIP' appendix when the review is ready.

Commit:
Add default custom theme in demo
modify visibility activity extension

Commit:
Remove NaraeImagePickerTheme from style.

also remove custom image picker theme from demo and replace with demo's app theme.

WindSekirun commented:
Released as 1.7.0. Thanks!
